app/scripts/views/settings/clients.js

```diff
@@ -115,41 +136,52 @@ define(function (require, exports, module) {
       });
     },
 
-    _onDeviceAdded: function () {
+    _onItemAdded: function () {
       this.render();
     },
 
-    _onDeviceRemoved: function (device) {
-      var id = device.get('id');
-      var self = this;
-      $('#' + id).slideUp(DEVICE_REMOVED_ANIMATION_MS, function () {
-        // re-render in case the last device is removed and the
-        // "no registered devices" message needs to be shown.
-        self.render();
-      });
+    _onItemRemoved: function (item) {
+      $('#' + item.get('id')).slideUp(DEVICE_REMOVED_ANIMATION_MS);
     },
 
-    _onDisconnectDevice: function (event) {
-      this.logViewEvent('disconnect');
-      var deviceId = $(event.currentTarget).attr('data-id');
-      this._destroyDevice(deviceId);
+    _onDisconnectClient: function (event) {
+      var itemId = $(event.currentTarget).data('id');
+      // type of client that was disconnected, can be 'client' or 'device'.
+      var clientType = $(event.currentTarget).data('type');
+
+      this.logViewEvent(clientType + '.disconnect');
+      if (clientType === 'device') {
```

Comment:
Should `device` and `app` (with the above caveats of app being overloaded) be extracted into Constants.js?